kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ConsumerGroupLagManagerImpl.java

```diff
@@ -0,0 +1,118 @@
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.controllers;
+
+import static io.confluent.kafkarest.controllers.Entities.checkEntityExists;
+import static java.util.Objects.requireNonNull;
+
+import io.confluent.kafkarest.entities.ConsumerGroup;
+import io.confluent.kafkarest.entities.ConsumerGroupLag;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import javax.inject.Inject;
+import javax.ws.rs.NotFoundException;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ListOffsetsResult.ListOffsetsResultInfo;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class ConsumerGroupLagManagerImpl extends AbstractConsumerLagManager implements ConsumerGroupLagManager {
+
+  private final ConsumerGroupManager consumerGroupManager;
+  private static final Logger log = LoggerFactory.getLogger(ConsumerGroupLagManagerImpl.class);
+
+  @Inject
+  ConsumerGroupLagManagerImpl(
+      Admin kafkaAdminClient,
+      ConsumerGroupManager consumerGroupManager) {
+    super(kafkaAdminClient);
+    this.consumerGroupManager = requireNonNull(consumerGroupManager);
+  }
+
+  @Override
+  public CompletableFuture<Optional<ConsumerGroupLag>> getConsumerGroupLag(
+      String clusterId,
+      String consumerGroupId
+  ) {
+    return consumerGroupManager.getConsumerGroup(clusterId, consumerGroupId)
+        .thenApply(
+            consumerGroup ->
+                checkEntityExists(consumerGroup, "Consumer Group %s could not be found.", consumerGroupId))
+        .thenCompose(
+            consumerGroup ->
+                getCurrentOffsets(consumerGroupId)
+                    .thenApply(
+                        fetchedCurrentOffsets ->
+                            checkOffsetsExist(fetchedCurrentOffsets, "Consumer group offsets could not be found."))
+                    .thenCompose(
+                        fetchedCurrentOffsets ->
+                            getLatestOffsets(fetchedCurrentOffsets)
+                                .thenApply(
+                                    latestOffsets ->
+                                       createConsumerGroupLag(
+                                           clusterId,
+                                           consumerGroup,
+                                           fetchedCurrentOffsets,
+                                           latestOffsets))));
+  }
+
+  Optional<ConsumerGroupLag> createConsumerGroupLag(
```

Comment:
This method never returns `Optional.empty()`, so it should probably returns `ConsumerGroupLag` instead of `Optional<ConsumerGroupLag>`. Also, it should be private and static.